src/drivers/stm32/drv_io_timer.c

```diff
@@ -486,6 +488,19 @@ static inline void io_timer_set_oneshot_mode(unsigned timer)
 	rEGR(timer) = GTIM_EGR_UG;
 }
 
+void io_timer_set_dshot_mode(unsigned channel, unsigned dshot_pwm_freq)
+{
+	unsigned timer = channels_timer(channel);
+
+	rARR(timer)  = DSHOT_MOTOR_PWM_BIT_WIDTH;
+	rPSC(timer)  = ((int)(io_timers[timer].clock_freq / dshot_pwm_freq)/DSHOT_MOTOR_PWM_BIT_WIDTH) - 1;;
```

Comment:
```suggestion
	rPSC(timer)  = ((int)(io_timers[timer].clock_freq / dshot_pwm_freq)/DSHOT_MOTOR_PWM_BIT_WIDTH) - 1;
```

Reply:
With calling init function with different DSHOT frequencies.

example:
```
up_dshot_init(channel_mask, DSHOT1200);
up_dshot_init(channel_mask, DSHOT600);
up_dshot_init(channel_mask, DSHOT300);
up_dshot_init(channel_mask, DSHOT150);
```

And then sending the arming command:
`up_dshot_arm(true);`

Setting throttle value:
```
    uint16_t throttle = 100;
    up_dshot_motor_data_prepare(0,throttle);
```

And triggering the message:
`up_dshot_trigger();`

I've connected logic analyzer on dshot (servo wire) to see frequencies. 
ESC was working for my setup Pixracer at 168 MHz on all Dshots modes. It was able to arm and to spin the motors. 

For each Dshot message, frequency needs to be near selected. 
For example dshot 1200, the frequency will be around 1.2 MHz. It depends on the selected system frequency, and this formula will calculate the closest one by rounding it down because of the integer.